src/rstar.jl

```diff
@@ -0,0 +1,125 @@
+
+# Keyword Arguments
+* `subset = 0.8` ... Subset used to train the classifier, i.e. 0.8 implies 80% of the samples are used.
+* `iterations = 10` ... Number of iterations used to estimate the statistic. If the classifier is not probabilistic, i.e. does not return class probabilities, it is advisable to use a value of one.
+* `verbosity = 0` ... Verbosity level used during fitting of the classifier.
+
+# Usage
+
+```julia-repl
+using MLJ, MLJModels
+# You need to load MLJBase and the respective package your are using for classification first.
+
+# Select a classification model to compute the Rstar statistic.
+# For example the XGBoost classifier.
+model = @load XGBoostClassifier()
+
+# Compute 100 samples of the R* statistic using sampling from according to the prediction probabilities.
+Rs = rstar(chn, model, iterations = 20)
+
+# estimate Rstar
+R = mean(Rs)
+
+# visualize distribution
+histogram(Rs)
+```
+
+## References:
+[Lambert & Vehtari] Ben Lambert and Aki Vehtari. "R∗: A robust MCMC convergence diagnostic with uncertainty using gradient-boostined machines." Arxiv 2020.
+"""
+function rstar(rng::Random.AbstractRNG, classif::MLJModelInterface.Supervised, x::RStarTable, y::AbstractVector{Int}; iterations = 10, subset = 0.8, verbosity = 0)
+
+    size(x,1) != length(y) && throw(DimensionMismatch())
+    iterations >= 1 && ArgumentError("Number of iterations has to be positive!")
+
+    if iterations > 1 && classif isa Deterministic
+        @warn("Classifier is not a probabilistic classifier but number of iterations is > 1.")
+    elseif iterations == 1 && classif isa Probabilistic
+        @warn("Classifier is probabilistic but number of iterations is equal to one.")
+    end
+
+    N = length(y)
+    K = length(unique(y))
+
+    # randomly sub-select training and testing set
+    Ntrain = round(Int, N*subset)
+    Ntest = N - Ntrain
+
+    ids = Random.randperm(rng, N)
+    train_ids = view(ids, 1:Ntrain)
+    test_ids = view(ids, (Ntrain+1):N)
+
+    # train classifier using XGBoost
+    fitresult, _ = MLJModelInterface.fit(classif, verbosity, x[train_ids], MLJModelInterface.categorical(y[train_ids]))
+
+    Rstats = fill(Inf, iterations)
+    xtest = x[test_ids]
+    ytest = MLJModelInterface.categorical(view(y, test_ids))
+
+    for i in 1:iterations
+
+        # predict labels for "test" data
+        p = MLJModelInterface.predict(classif, fitresult, xtest)
+        pred = similar(ytest)
+
+        if classif isa Deterministic
+            pred[:] = p
+        elseif classif isa Probabilistic
+            pred[:] = get.(rand.(rng, p))
+        else
+            throw(ErrorException("Unknown type of classifier"))
+        end
```

Comment:
I guess it would be easier to not preallocate `pred` but compute it directly:
```suggestion
        if classif isa Deterministic
            pred = MLJModelInterface.predict(classif, fitresult, xtest)
        elseif classif isa Probabilistic
            pred = get.(rand.(rng, MLJModelInterface.predict(classif, fitresult, xtest)))
        end
```
Also the type of classifier can be checked outside of the loop.